Src/FluentAssertions/Primitives/DateTimeAssertions.cs

```diff
@@ -715,6 +715,62 @@ public AndConstraint<TAssertions> NotHaveSecond(int unexpected,
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Asserts that the current <see cref="DateTime"/>  has the <paramref name="expected"/> millisecond.
+    /// </summary>
+    /// <param name="expected">The expected milliseconds of the current value.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<TAssertions> HaveMillisecond(int expected,
+        [StringSyntax("CompositeFormat")] string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .WithExpectation("Expected the milliseconds part of {context:the time} to be {0}{reason}", expected)
+            .ForCondition(Subject.HasValue)
+            .FailWith(", but found a <null> DateTime.")
+            .Then
+            .ForCondition(Subject.Value.Millisecond == expected)
```

Comment:
🔧 My IDE tells me that `Subject.Value` can be `null`. You can suppress that with `!`